pandas/tests/series/test_apply.py

```diff
@@ -576,3 +576,13 @@ def f(x):
         result = s.map(f)
         exp = pd.Series(['Asia/Tokyo'] * 25, name='XX')
         tm.assert_series_equal(result, exp)
+
+    @pytest.mark.parametrize("mapping,exp", [
+        ({np.nan: 'not NaN'}, pd.Series(['not NaN'])),
+        ({'string': 'another string' }, pd.Series(['another string'])),
+        ({42: 'the answer'}, pd.Series(['the answer']))])
+    def test_map_missing_mixed(self, mapping, exp):
+        s = pd.Series(list(mapping.keys())[0])
```

Comment:
Nice job on the parametrization but in the process of updating this line I think we are losing sight of what we are trying to test. Originally you were constructing this series from `list('abcd')` and appending an NA record. Now all you are doing is using the key of your dict as the series value, but that's not the same test.

Now that I'm thinking about it it's probably good to also add a "vals" parameter in front of the mapping. For the first two scenarios you can use `list('abcd')` as you had before and for the third use `range(4)`.  Then just make your first line `s = pd.Series(vals + [np.nan])`

Reply:
OK, I'll give that a shot.